Add docs for sysvar cache (#5563)
pingcap/docs: system-variables.md

> **Note:**
>

> > <!-- Since next_version version --> Executing `SET GLOBAL` applies immediately on the TiDB server where the statement was issued. All TiDB servers in the cluster are sent a notification to refresh their global variables, which will begin immediately as a background operation.

Reviewer: `All other TiDB servers`, `receive a notification`, maybe?

BTW, I think it should be noted that the notification is immediate, but the refresh/eviction is not.

Author: It's actually not "all other servers" but "all servers" (the initiating server will also receive the message too).

I used the language "begin immediately as a background operation" to indicate that it has no guarantee on completion time.

Author: @xhebox @tiancaiamao I have updated the wording. PTAL again, thx!

> > Since TiDB 5.1, executing `SET GLOBAL` applies immediately on the TiDB server where the statement was issued. A notification is then sent to all TiDB servers to refresh their system variable cache, which will start immediately as a background operation. Because there is a risk that some TiDB servers may miss the notification, the system variable cache is also refreshed automatically every 30 seconds. This helps ensure that all servers are operating with the same configuration.

Reviewer 2: We can add "Since TiDB 5.1" back in the upcoming release-5.1 branch, which will be created the next week. 
```suggestion
> Executing `SET GLOBAL` applies immediately on the TiDB server where the statement was issued. A notification is then sent to all TiDB servers to refresh their system variable cache, which will start immediately as a background operation. Because there is a risk that some TiDB servers might miss the notification, the system variable cache is also refreshed automatically every 30 seconds. This helps ensure that all servers are operating with the same configuration.
```

> > <!-- Since next_version version --> Executing `SET GLOBAL` applies immediately on the TiDB server where the statement was issued. A notification is then sent to all TiDB servers to refresh their system variable cache, which will start immediately as a background operation. Because there is a risk that some TiDB servers may miss the notification, the system variable cache is also refreshed automatically every 30 seconds. This helps ensure that all servers are operating with the same configuration.

Reviewer 2: Does the `next_version version` mean v5.1?
```suggestion
> <!-- Since next_version version --> Executing `SET GLOBAL` applies immediately on the TiDB server where the statement was issued. A notification is then sent to all TiDB servers to refresh their system variable cache, which will start immediately as a background operation. Because there is a risk that some TiDB servers might miss the notification, the system variable cache is also refreshed automatically every 30 seconds. This helps ensure that all servers are operating with the same configuration.
```

Author: Yes, it means since 5.1.

Reviewer 2: What about adding "Since v5.1" in the upcoming release-5.1 branch (not created yet) and leave no version information in the master branch?

>

> > TiDB differs from MySQL in that `GLOBAL` scoped variables **persist** through TiDB server restarts. Additionally, TiDB presents several MySQL variables as both readable and settable. This is required for compatibility, since it is common for both applications and connectors to read MySQL variables. For example: JDBC connectors both read and set query cache settings, despite not relying on the behavior.

Reviewer 2: ```suggestion
> TiDB differs from MySQL in that `GLOBAL` scoped variables **persist** through TiDB server restarts. Additionally, TiDB presents several MySQL variables as both readable and settable. This is required for compatibility, because it is common for both applications and connectors to read MySQL variables. For example, JDBC connectors both read and set query cache settings, despite not relying on the behavior.
```
